File: Dockerfile.go
# 使用 Go 镜像
FROM golang:1.19.3 AS go-builder

WORKDIR /app

ENV GOPROXY=https://goproxy.cn

# 复制 go.mod 和 go.sum 文件并下载依赖
COPY go.mod go.sum ./
RUN go mod download

# 复制整个项目并构建
COPY . .

# 构建 Go 项目
RUN go build -o main ./server

# 最终镜像
FROM golang:1.19.3

WORKDIR /app

# 复制二进制文件
COPY --from=go-builder /app/main /app/main

# 复制其他需要的文件
COPY server/template /app/server/template
COPY server/pb /app/server/template
COPY persist /app/persist
COPY util /app/util

# 启动 Go 项目
CMD ["./main"]

File: Dockerfile.py
# 使用 Python 3.6.9 镜像作为基础镜像
FROM python:3.6.9

# 设置工作目录
WORKDIR /app

# 添加当前目录到 Python 路径中
ENV PYTHONPATH /app/crawler

# 复制项目文件到容器中
COPY ./crawler /app/crawler

# 安装 Python 依赖
RUN pip install --no-cache-dir -r /app/crawler/requirements.txt

# 暴露所需的端口（如果有需要）
# EXPOSE 5000

# 定义容器启动命令
CMD ["python", "-m", "crawler.entrance"]